### ISSUE TITLE
Unable to produce repeating controls and attributes for fields in multiple stages in same BPF

### ISSUE THREAD

**Issue by TomMalow** [delegateas/XrmDefinitelyTyped#77]
If a BPF has multiple stages with the same field appearing in multiple stages. Then the field will have different schema names for each field in each stage. XDT only generates the getControl and getAttribute overload methods for the first header field that appears in the BPF.

For example, If you have a BPF with three stages and the same field, "my_field" appearing all of the stages and the field "my_additional_field" in the last two. Then the fields will have the following schema name each stage:
**Stage1**
`header_process_my_field`

**Stage2**
`header_process_my_field_1`
`header_process_my_additional_field`

**Stage3**
`header_process_my_field_2`
`header_process_my_additional_field_1`

XDT will only generate the overload methods for `header_process_my_field` and `header_process_my_additional_field` and you are unable to work with the remaining fields in the BPF.

**Comment by magesoe** [comment]
Added in version 4.1.0

**Comment by henrikhannemose** [comment]
When looking at the output we generate with the latest version of XDT, it appears that we create too many controls that do not exist. By going back to older versions this appears to have been introduced in XDT 4.1.0 (see the following excerpt from `interface Controls extends Xrm.ControlCollectionBase`):

**XDT 4.0.9:**
```
get(name: "header_ownerid"): Xrm.LookupControl<"systemuser" | "team">;
get(name: "header_process_actualclosedate"): Xrm.DateControl;
get(name: "header_process_currentsituation"): Xrm.StringControl;
get(name: "header_process_customerneed"): Xrm.StringControl;
get(name: "header_process_description"): Xrm.StringControl;
```

**XDT 4.1.0:**
```
get(name: "header_ownerid"): Xrm.LookupControl<"systemuser" | "team">;
get(name: "header_process_actualclosedate"): Xrm.DateControl | null;
get(name: "header_process_actualclosedate1"): Xrm.DateControl | null;
get(name: "header_process_actualclosedate2"): Xrm.DateControl | null;
get(name: "header_process_actualclosedate3"): Xrm.DateControl | null;
get(name: "header_process_actualclosedate4"): Xrm.DateControl | null;
get(name: "header_process_actualclosedate5"): Xrm.DateControl | null;
get(name: "header_process_actualclosedate6"): Xrm.DateControl | null;
get(name: "header_process_currentsituation"): Xrm.StringControl | null;
get(name: "header_process_customerneed"): Xrm.StringControl | null;
get(name: "header_process_customerneed1"): Xrm.StringControl | null;
get(name: "header_process_description"): Xrm.StringControl | null;
```
In this specific case, I have `actualclosedate` on the form once and in the BPF once.

a) As far as I can see, the output from XDT 4.0.9 is the correct output (i.e. we generate too many controls with numbers appended today).
b) For fields that are actually added multiple times to the BPF, I think the appended number should contain an underscore (i.e. *_1, not just *1).

**Comment by Dashulol** [comment]
I ran into an issue with this. While the fields did get generated, the naming schema was wrong. I'm using CRM 8.2 and XDT 5.0.0.

Expected: header_process_myfield_2
Actual: header_process_myfield2

**Comment by obkirneh** [comment]
I have the same issue with same field multiple times in BPF. 
Here is an example:

> getControl(controlName: "header_process_rm_estclosedate"): Xrm.DateControl | null;
> getControl(controlName: "header_process_rm_estclosedate1"): Xrm.DateControl | null;

Should be 

> getControl(controlName: "header_process_rm_estclosedate"): Xrm.DateControl | null;
> getControl(controlName: "header_process_rm_estclosedate_1"): Xrm.DateControl | null;

Our current version of XrmDefinitelyTyped is: 4.4.0

Note: _Today (2020.02.12) updated to v5.1 of XrmDefinitelyTyped and still got same issue._

**Comment by magesoe** [comment]
Added fixes in 5.1.6
